src/win/tty.c

```diff
@@ -57,11 +57,17 @@
 
 static void uv_tty_capture_initial_style(CONSOLE_SCREEN_BUFFER_INFO* info);
 static void uv_tty_update_virtual_window(CONSOLE_SCREEN_BUFFER_INFO* info);
+static int uv__cancel_read_console(uv_tty_t* handle);
 
 
 /* Null uv_buf_t */
 static const uv_buf_t uv_null_buf_ = { 0, NULL };
 
+static CRITICAL_SECTION uv__read_console_trap_lock;
+static volatile BOOL uv__read_console_trap = FALSE;
```

Comment:
You are probably right, but in my (long, possibly dated) experience with VC++ it's safer to rely on explicit volatile declarations. It should be noted that the volatile semantics are MS-specific: https://msdn.microsoft.com/en-us/library/12a04hfd.aspx.
It's possible that the compiler has gotten smarter and knows not to reorder accesses around critical sections, but is there really much harm in making it explicit?